src/DataGrid.tsx

```diff
@@ -572,11 +572,12 @@ function DataGrid<R, SR, K extends Key>(
 
   function updateRow(rowIdx: number, row: R) {
     if (typeof onRowsChange !== 'function') return;
-    if (row === rawRows[rowIdx]) return;
+    const rawRowIdx = getRawRowIdx(rowIdx);
```

Comment:
I moved this here as `updateRow` is used by `formatter.onRowChange` also. It also fixes target row index while pasting